packages/frontend/core/src/modules/peek-view/view/modal-container.tsx

```diff
@@ -0,0 +1,127 @@
+import * as Dialog from '@radix-ui/react-dialog';
+import { cssVar } from '@toeverything/theme';
+import { assignInlineVars } from '@vanilla-extract/dynamic';
+import {
+  createContext,
+  type PropsWithChildren,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+import useTransition from 'react-transition-state';
+
+import * as styles from './modal-container.css';
+
+const animationTimeout = 200;
+
+const contentOptions: Dialog.DialogContentProps = {
+  ['data-testid' as string]: 'peek-view-modal',
+  onPointerDownOutside: e => {
+    const el = e.target as HTMLElement;
+    if (el.closest('[data-peek-view-wrapper]')) {
+      e.preventDefault();
+    }
+  },
+  style: {
+    padding: 0,
+    backgroundColor: cssVar('backgroundPrimaryColor'),
+    overflow: 'hidden',
+  },
+};
+
+// a dummy context to let elements know they are inside a peek view
+export const PeekViewContext = createContext<Record<string, never> | null>(
+  null
+);
+
+const emptyContext = {};
+
+export const useInsidePeekView = () => {
+  const context = useContext(PeekViewContext);
+  return !!context;
+};
+
+function getElementScreenPositionCenter(target: HTMLElement) {
+  const rect = target.getBoundingClientRect();
+
+  if (rect.top === 0 || rect.left === 0) {
+    return null;
+  }
+
+  const scrollTop = window.scrollY || document.documentElement.scrollTop;
+  const scrollLeft = window.scrollX || document.documentElement.scrollLeft;
+
+  return {
+    x: rect.x + scrollTop + rect.width / 2,
```

Comment:
Why to add the two values in the cross-axis direction? Shouldn't it be `rect.x + scrollLeft`?

Reply:
sorry, typo